.github/workflows/pre-submit.lint.yml

```diff
@@ -0,0 +1,79 @@
+name: Lint
+
+on: [pull_request]
+
+permissions:
+  # Needed to check out the repo.
+  contents: read
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      - uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923 # v3.2.0
+        with:
+          go-version: "1.18"
+      - env:
+          GOLANGCI_LINT_VERSION: "1.46.2"
+        run: |
+          set -euo pipefail
+
+          #Install golangci-lint
+          wget -qO- "https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz" | tar -xzf -
```

Comment:
The issue with the `wget` approach is that there isn't a check for the SHA? Also, this approach can't keep track of updates for golangci-lint package.

The approach scorecard has taken is to have all the tools via a separate tools package and https://github.com/ossf/scorecard/blob/main/tools/tools.go and installing it in CI

Reply:
Or another approach is to use the https://github.com/golangci/golangci-lint-action

Reply:
I added a checksum check for 1.46.2. Checksum is included here since attackers that can modify the release can modify the checksum file as well.

To be fair, most GitHub actions don't actually do this. For example the golangci-lint-action [just installs](https://github.com/golangci/golangci-lint-action/blob/master/src/install.ts#L34-L60) the tar gz and uses the binary.

Reply:
Why not use the `tools` approach with a separate `go.mod`? Now the dep's can be out of sync with the version and it is hard to maintain with this approach.

Reply:
What I meant is that? How are we going to get updates for this dependency?

The goal is not be be out of sync.

Reply:
I'm not sure what you mean by being out of sync. Out of sync with the latest version of golangci-lint?

My intention is that when we need to use a new version of golangci-lint we will update the version and the sha in this file. That's intentional as I don't think we need (or want) to update it automatically.

Reply:
After thinking a sec, I think perhaps that you mean that if we use an action then dependabot will update the dependency for us?

I agree that that would be helpful, but I'm not sure I want to use a 3rd party action simply for that reason.